src/hooks/use-contract-activity.ts

```diff
@@ -51,31 +68,54 @@ export const useContractActivity = (tokenSlug?: string): UseActivityInterface =>
   };
 
   const handleUpdate = async () => {
-    if (activities.length > 0 && !isAllLoaded) {
-      const lastActivityGroup = activities[activities.length - 1].sort((a, b) => b.id - a.id);
+    let newActivities: Array<ActivityGroup> = [];
+    const wasLoading = isLoading;
```

Comment:
mb `if (isLoading) return;` ?

Reply:
I think it's not the case, we may need to update `lastActivityRef` value